gh-129149: Add Missing fast path in PYLONG_FROM_UINT macro for compact integers (#129168)
python/cpython: Misc/NEWS.d/next/Core_and_Builtins/2025-01-22-14-24-44.gh-issue-129149.wAYu43.rst

> Add Missing fast path in PyLong_FromLong(), PyLong_FromLongLong() functions

Reviewer: ```suggestion
Add missing fast path in :func:`PyLong_FromLong` and :func:`PyLong_FromLongLong` functions
```

> Add missing fast path for medium-size integers in :c:func:`PyLong_FromUnsignedLong`,

Reviewer 2: ```suggestion
Add fast path for medium-size integers in :c:func:`PyLong_FromUnsignedLong`,
```

> Add missing fast path for medium-size integers  in :c:func:`PyLong_FromUnsignedLong`,

Reviewer 3: ```suggestion
Add missing fast path for medium-size integers in :c:func:`PyLong_FromUnsignedLong`,
```

> Add missing fast path in :func:`PyLong_FromLong` and :func:`PyLong_FromLongLong` functions

Reviewer 4: ```suggestion
Add missing fast path in :c:func:`PyLong_FromLong` and :c:func:`PyLong_FromLongLong` functions
```

> Add missing fast path in :c:func:`PyLong_FromLong` and :c:func:`PyLong_FromLongLong` functions

Reviewer 5: The fast path already existed in those two functions, it was just refactored into PYLONG_FROM_SIGNED.

Since you've added the fast path to `PYLONG_FROM_UINT`, `PyLong_FromUnsignedLong()`, `PyLong_FromUnsignedLongLong()` and `PyLong_FromSize_t()` got it now.

It does not (yet) exist in `PyLong_FromSsize_t()`, where the issue started from in https://github.com/python/cpython/pull/128927#discussion_r1924265225.

Author: Addressed in [37a5c9a](https://github.com/python/cpython/pull/129168/commits/37a5c9aa3bb0bca09071170032244c7ba7faaa00)

> for compact integers

Reviewer: ```suggestion
for compact integers.
```

> Add missing fast path in :c:func:`PyLong_FromUnsignedLong`, :c:func:`PyLong_FromUnsignedLongLong` and :c:func:`PyLong_FromSize_t` functions
> for compact integers.

Reviewer 4: ```suggestion
Add missing fast path for medium-size integers in :c:func:`PyLong_FromLong`.
```

Author: `PyLong_FromLong` already contains fast path for medium-size integers https://github.com/python/cpython/blob/main/Objects/longobject.c#L329-L330

Author: I have addressed `for medium-size integers` [5760c95](https://github.com/python/cpython/pull/129168/commits/5760c954fd39b9787a6616db4efe967a67d6ef39)

> for compact integers. And refactor :c:func:`PyLong_FromLong` and :c:func:`PyLong_FromLongLong` to use the macro ``PYLONG_FROM_SIGNED``.

Reviewer 3: I'm not sure it's needed to know that the functions were refactored using a private macro. The code was only moved around right?

Reviewer 4: I suspect that the change might affect performance.  That's something that should be documented.  We shouldn't mention the private macro.

Reviewer 3: I'm not saying we should remove the mention to the missing fast path. I'm suggesting removing the refactoring information.

Reviewer 3: And also revert the refactoring if it does not add new value to the current code. If the current code is already having a fast path, I would prefer doing it two different commits (easier to bisect in case of an issue)
